File: Week_01/reverseList.c
#include <stdio.h>

//Definition for singly-linked list.
//struct ListNode {
//	int val;
//	struct ListNode *next;
//};

//思路：使用双指针，感觉链表题需要注意的就是指针的顺序，保证迭代不会中断
struct ListNode* reverseList(struct ListNode* head) {
	struct ListNode *prev = NULL;
	struct ListNode *current = head;
	while (current != NULL) {
		struct ListNode *tempNode = current->next;
		current->next = prev;
		prev = current;
		current = tempNode;
	}
	return prev;
}

int main(int argc, char *argv[]) {
	
}

File: Week_01/reverse_list.rb
# Definition for singly-linked list.
class ListNode
    attr_accessor :val, :next
    def initialize(val)
        @val = val
        @next = nil
    end
end

# @param {ListNode} head
# @return {ListNode}
def reverse_list(head)
    prev = nil
    curr = head
    until curr.nil?
        temp = curr.next
        prev = curr
        curr = temp
    end
    return curr
end